Iteration 1: User story 5

dhalverson opened the pull request:
**Adds**

- User model tests
- Updates nav bar to include admin view logic
- Adds admin routes
- Creates admin dashboard and user HTML placeholders
- Adds enum roles to user
- Creates admin controller

dhalverson committed:
Staging work for US5

dhalverson committed:
Add default, merchant, and admin tests

dhalverson committed:
Adds admin and non admin view logic to nav bar

dhalverson committed:
Adds admin routes

dhalverson committed:
Adds view logic if admin to nav bar

dhalverson committed:
Adds admin view placeholders

dhalverson committed:
Adds enum roles

dhalverson committed:
Adds helper method, current user and current admin method

dhalverson committed:
Seeds with users

dhalverson committed:
Creates controller

dhalverson committed:
Merge branch 'development' into user_story_5

dhalverson commented on spec/features/navigation_spec.rb:
```diff
@@ -43,4 +43,36 @@
       end
     end
   end
+  describe 'As an Admin' do
```
Need to add other types of users to this test for edge cases

dhalverson replied:
Need to test edge case here, to test other types of users

dhalverson commented on spec/models/user_spec.rb:
```diff
@@ -12,4 +12,49 @@
     it { should validate_presence_of :password }
   end
 
+  describe "roles" do
```
Need to test methods here as well

dhalverson replied:
Need to test the methods here

dhalverson reviewed (commented):
Need to update feature and model tests for this PR

PhilipDeFraties commented on app/controllers/admins_controller.rb:
```diff
@@ -0,0 +1,10 @@
+class AdminsController < ApplicationController
```
Admin::DashboardController < Admin::BaseController I think will be the first line

PhilipDeFraties reviewed (changes requested):
Looks good, I think you did the migration correctly. Just gotta look at the admin controllers and figure out the base/dashboard stuff.

PhilipDeFraties commented on config/routes.rb:
```diff
@@ -40,4 +40,7 @@
   post "/users/new", to: "users#create"
   get "/users/:id", to: "users#show"
+
+  get '/admin', to: 'admins#dashboard'
```
I think the routes here are a bit off, I dont think dashboard is supposed to be an action, I believe the route would look like this: 
`get '/admin/dashboard', to: "dashboard#index"`
This will then look for a controller called Admin::DashboardController

PhilipDeFraties committed:
Delete application_controller.rb

The changes Daniel and I made for this file I think are cleanest so Im just removing this file from the pr

PhilipDeFraties committed:
Delete schema.rb

These changes were already implemented with the previous pr so im removing

PhilipDeFraties committed:
Delete 20200909235921_add_role_to_users.rb

Changes already in made with previous pr so I'm removing

PhilipDeFraties committed:
Delete user.rb

Changes already implemented in previous pr so Im removing

dhalverson commented:
Awaiting US 8 PR to add this content
